src/monitors.ts

```diff
@@ -244,16 +246,24 @@ export function replaceCriticalThreshold(query: string, criticalThreshold: numbe
 }
 
 /**
- * Helper function that interpolates template variables, e.g. replaces template
- * variable `$functionName` with its default value `{{functionname.name}}`.
- * @param templateVariables - Template variables defined in the monitor JSON
- * @param name - The string that may contain uninterpolated template variables,
+ * Helper function that interpolates template variables (if any), e.g.
+ * replaces template variable `$functionName` with its default value
+ * `{{functionname.name}}`. Template variables with no default value will be
+ * ignored.
+ * @param recommendedMonitors  - recommended monitors
+ * @param baseString - The string that may contain uninterpolated template variables,
  *        e.g. "High Error Rate on $functionName in $regionName for $awsAccount"
  * @returns Interpolation result, e.g. "High Error Rate on {{functionname.name}} in {{region.name}} for {{aws_account.name}}"
  */
-function interpolateTemplateVariables(templateVariables: TemplateVariable[], name: string): string {
-  templateVariables.forEach((templateVariable) => {
-    name = name.replace("$" + templateVariable.name, templateVariable.defaults[0]);
-  });
-  return name;
+function interpolateTemplateVariables(recommendedMonitor: ServerlessMonitor, baseString: string): string {
+  const templateVariables = recommendedMonitor.templateVariables;
+  let interpolatedString = baseString;
+  if (templateVariables !== undefined && templateVariables.length > 0) {
+    templateVariables.forEach((templateVariable) => {
+      if (templateVariable.defaults.length > 0) {
+        interpolatedString = interpolatedString.replace("$" + templateVariable.name, templateVariable.defaults[0]);
+      }
```

Comment:
I'm curious now 🤔 . Why does the monitors team have defaults defined as an array for a template variable? What does it mean when there are multiple values in this array? We are always taking just the first value and consuming it. It would be better to understand their intention so that we handle it appropriately.

Reply:
Good point. I guess an empty array means no default value, an array with one element means default value exists, and all the elements after the first one are ignored.

Confirming with them:
https://datadoghq.atlassian.net/wiki/spaces/AP/pages/3230303531/Product-based+Recommended+Monitors+Usage+Guide?focusedCommentId=3948118079